test/run_test.py

```diff
@@ -220,30 +235,50 @@ def get_python_command(options):
         return os.environ.get('PYCMD', 'python')
 
 
+def find_test_index(test, selected_tests, find_last_index=False):
```

Comment:
Why do we need this `find_last_index` behavior?

Reply:
The last index is needed if we have a list of consequent same test module tests to run, e.g.:

python run_test.py -i autograd cuda **torch.TestTorch.test_acos torch.TestTorch.test_tan torch.TestTorch.test_add** utils -f cuda -l torch

Initially ```selected_tests``` will look like this ```['autograd', 'cuda', 'torch.TestTorch.test_acos', 'torch.TestTorch.test_tan', 'torch.TestTorch.test_add', 'utils']```. When ```options.first``` is processed it will return an index of 1 and ```selected_tests``` will be updated to ```['cuda', 'torch.TestTorch.test_acos', 'torch.TestTorch.test_tan', 'torch.TestTorch.test_add', 'utils']``. After ```options.last``` is processed it should find the last index of the torch* tests which will be 3 in this case and the result should be ```['cuda', 'torch.TestTorch.test_acos', 'torch.TestTorch.test_tan', 'torch.TestTorch.test_add']```

I don't think this scenario should be very common as **-f** and **-l** more likely to be used to specify subset of the default tests and not be used together with -i where now we can specify multiple tests part of the same test module but it's good to support such behaviour.

Reply:
I see. That makes sense. I would write something like this as the docstring.

Reply:
sure